hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java

```diff
@@ -41,6 +55,19 @@ public DiscoveryJoiner(Node node, DiscoveryService discoveryService, boolean use
     }
 
     @Override
+    protected Collection<Address> getPossibleAddressesForInitialJoin() {
```

Comment:
This method name is misleading to me, since it's impossible to figure out it's gonna wait for addresses.
Shouldn't it be sth along these lines `getOrWaitForInitialJoinAddresses()`?

Reply:
Similar to [`WanBatchReplication#awaitAndGetTargetEndpoints`](https://github.com/hazelcast/hazelcast-enterprise/blob/e6d82a33b52c16369fa9e58d4618ecccfb30daf4/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/replication/WanBatchReplication.java#L112-L117) ?

Reply:
the thing is the waiting is just one of the possible strategies. 

the caller indicates its intend ("I am doing the initial join") and it's up to the implementation to choose what's sensible strategy. 

for example the `TCPJoiner` does not wait. It'll just try to connect (as it already has a list of possible targets in its configuration). while the `DiscoveryJoiner` has to wait.